crates/resolver/src/name_server/name_server_pool.rs

```diff
@@ -86,9 +93,62 @@ impl<P: ConnectionProvider> DnsHandle for NameServerPool<P> {
 
     fn send(&self, request: DnsRequest) -> Self::Response {
         let state = self.state.clone();
+        let acs = self.state.cx.answer_address_filter.clone();
         Box::pin(once(async move {
             debug!("sending request: {:?}", request.queries());
-            state.try_send(request).await
+            let mut response = state.try_send(request).await?;
+
+            let bind = response.owned_queries();
+            let Some(query) = bind.first() else {
+                return Err("no query in response".into());
+            };
```

Comment:
Would be nice to avoid cloning unless it's actually necessary, which is only after `bind.first()` has been checked, right?